test/response_test.rb

```diff
@@ -1461,14 +1461,19 @@ def generate_audience_error(expected, actual)
         end
       end
 
-      it 'raise if an encrypted assertion is found and the sp private key is wrong' do
+      it 'raise if an encrypted assertion is found and the SP private key does not match cert' do
         settings.certificate = ruby_saml_cert_text
-        wrong_private_key = ruby_saml_key_text.sub!('A', 'B')
+        wrong_private_key = ruby_saml_key_text.sub!('Z', 'X')
```

Comment:
This spec was kind of silly before, previously `ruby_saml_key_text.sub!('A', 'B')` was gsubbing `-----RSA PRIVATE KEY-----` to `RSB`, and not actually affecting the pkey body. I've improved it.